plugin/storage/integration/grpc_test.go

```diff
@@ -20,32 +20,39 @@ import (
 
 type GRPCStorageIntegrationTestSuite struct {
 	StorageIntegration
-	flags         []string
-	factory       *grpc.Factory
-	remoteStorage *RemoteMemoryStorage
+	flags          []string
+	factory        *grpc.Factory
+	archiveFactory *grpc.Factory
+	remoteStorage  *RemoteMemoryStorage
 }
 
 func (s *GRPCStorageIntegrationTestSuite) initialize(t *testing.T) {
 	logger := zaptest.NewLogger(t, zaptest.WrapOptions(zap.AddCaller()))
 	s.remoteStorage = StartNewRemoteMemoryStorage(t)
 
-	f := grpc.NewFactory()
-	v, command := config.Viperize(f.AddFlags)
-	err := command.ParseFlags(s.flags)
-	require.NoError(t, err)
-	f.InitFromViper(v, logger)
-	require.NoError(t, f.Initialize(metrics.NullFactory, logger))
+	initFactory := func() *grpc.Factory {
+		f := grpc.NewFactory()
+		v, command := config.Viperize(f.AddFlags)
+		err := command.ParseFlags(s.flags)
+		require.NoError(t, err)
+		f.InitFromViper(v, logger)
+		require.NoError(t, f.Initialize(metrics.NullFactory, logger))
+		return f
+	}
+	f := initFactory()
+	af := initFactory()
```

Comment:
so this creates a gRPC proxy pointing to the same remote storage port? I.e. not really testing two separate storage namespaces? Maybe we can fix in the follow-up.

Reply:
@yurishkuro Yeah I was actually thinking about this as well. The GRPC storage doesn't have a different namespace for archive so is it actually possible to configured grpc as archive in v1?

Reply:
we may have to add it. I don't see how otherwise we can make backwards compatible change with external implementations if they implement Archive capability of the Remote Storage API v1.